### ISSUE TITLE
"Illegal instruction" on raspbian wheezy

### ISSUE THREAD

**Issue by psych0d0g** [hzeller/gmrender-resurrect#25]
Compiled on my Pi and when i want to run the binary, all i get is a "Illegal instruction" error.
model B rev 2 PI

**Comment by tryptophan0** [comment]
Hi. I am receiving the same error message. 
This is the output from running gmediarender under gdb.
The hardware I am using is an Pi Zero W and pHAT DAC

```
pi@raspberrypi:~/gmrender-resurrect-master $ gdb src/gmediarender 
GNU gdb (Raspbian 7.12-6) 7.12.0.20161007-git
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from src/gmediarender...done.
(gdb) run
Starting program: /home/pi/gmrender-resurrect-master/src/gmediarender 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
gmediarender 0.0.7-git started [ gmediarender 0.0.7-git (libupnp-1.6.19+git20160116; glib-2.50.3; gstreamer-1.10.4) ].
Logging switched off. Enable with --logfile=<filename> (or --logfile=stdout for console)
[New Thread 0xb51b1470 (LWP 1801)]
[New Thread 0xb49b1470 (LWP 1802)]
[Thread 0xb49b1470 (LWP 1802) exited]
[New Thread 0xb49b1470 (LWP 1803)]

(gmediarender:1798): GStreamer-CRITICAL **: gst_pad_new_from_template: assertion 'GST_IS_PAD_TEMPLATE (templ)' failed

(gmediarender:1798): GStreamer-CRITICAL **: gst_pad_template_get_caps: assertion 'GST_IS_PAD_TEMPLATE (templ)' failed

** (gmediarender:1798): CRITICAL **: gst_pad_set_caps: assertion 'GST_IS_PAD (pad)' failed

(gmediarender:1798): GStreamer-CRITICAL **: gst_element_add_pad: assertion 'GST_IS_PAD (pad)' failed
ERROR [2019-01-30 18:38:37.139790 | webserver] Could not stat '/usr/local/share/gmediarender/grender-64x64.png': No such file or directory
ERROR [2019-01-30 18:38:37.141083 | webserver] Could not stat '/usr/local/share/gmediarender/grender-128x128.png': No such file or directory
[New Thread 0xa8342470 (LWP 1821)]
[New Thread 0xa7b42470 (LWP 1822)]
[New Thread 0xa7342470 (LWP 1823)]
[New Thread 0xa6b42470 (LWP 1824)]
[New Thread 0xa6342470 (LWP 1825)]
[New Thread 0xa5b42470 (LWP 1826)]
[New Thread 0xa5342470 (LWP 1827)]
[New Thread 0xa4b42470 (LWP 1828)]
[New Thread 0xa4342470 (LWP 1829)]
[New Thread 0xa3b42470 (LWP 1830)]
Ready for rendering.
[New Thread 0xa2bff470 (LWP 1831)]
[Thread 0xa7b42470 (LWP 1822) exited]
[Thread 0xa4b42470 (LWP 1828) exited]
[Thread 0xa5b42470 (LWP 1826) exited]
[New Thread 0xa5b42470 (LWP 1832)]
[New Thread 0xa4b42470 (LWP 1833)]
[New Thread 0xa7b42470 (LWP 1834)]
[Thread 0xa7b42470 (LWP 1834) exited]
[New Thread 0xa7b42470 (LWP 1835)]
[Thread 0xa7b42470 (LWP 1835) exited]
[New Thread 0xa23ff470 (LWP 1836)]
[Thread 0xa23ff470 (LWP 1836) exited]
[New Thread 0xa23ff470 (LWP 1837)]
[Thread 0xa23ff470 (LWP 1837) exited]
[New Thread 0xa7b42470 (LWP 1838)]
[Thread 0xa7b42470 (LWP 1838) exited]
[New Thread 0xa237f470 (LWP 1840)]
[Thread 0xa237f470 (LWP 1840) exited]
[New Thread 0xa7b42470 (LWP 1841)]
AL lib: (EE) UpdateDeviceParams: Failed to set 44100hz, got 48000hz instead
[New Thread 0xa237f470 (LWP 1842)]
[New Thread 0xa1aab470 (LWP 1843)]
[New Thread 0xa12ab470 (LWP 1844)]

Thread 25 "aqueue:src" received signal SIGILL, Illegal instruction.
[Switching to Thread 0xa237f470 (LWP 1842)]
0xaec2dad4 in ?? () from /usr/lib/arm-linux-gnueabihf/libopenal.so.1
(gdb) bt
#0  0xaec2dad4 in ?? () from /usr/lib/arm-linux-gnueabihf/libopenal.so.1
#1  0x00000800 in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) 
```

**Comment by hzeller** [comment]
I suspect one of the libraries was compiled for the Pi3, while the Pi Zero W has an older generation ARM with probably limited instruction set. In particular the gstreamer libraries probably use some optimized floating point operations that might not be available on the Pi.

I'd probably suggest to recompile gstreamer from scratch on your platform and not use the packages.

**Comment by tryptophan0** [comment]
Wow, that's a fast reply :)
From what you were suggesting, I suspected the culprit to be a part of the "bad" gstreamer1.0-plugins.
I removed the package and the error has not appeared again. It still wasn't working (presumably from me, messing with the system too much), so I reinstalled Raspbian. This time I have installed the gmediarender, gstreamer1.0-libav and gstreamer1.0-alsa via apt-get.
 gstreamer1.0-plugins-base, gstreamer1.0-plugins-good have installed automatically (but not bad and ugly!).
This time it works :tada:  Thanks a lot!
